src/TwitchLib.Communication/Models/ReconnectionPolicy.cs

```diff
@@ -1,93 +1,231 @@
+        ///     <b>infinitely</b>
+        ///     attempts to reconnect
+        ///     <br></br>
+        ///     <br></br>
+        ///     with each attempt, the reconnect interval increases by 3_000 milliseconds
+        ///     until it reaches 30_000 milliseconds
+        ///     <br></br>
+        ///     
+        ///     <br></br>
+        ///     <br></br>
+        ///     Example:
+        ///     <br></br>
+        ///     try to connect -> couldnt connect -> wait 3_000 milliseconds -> try to connect -> couldnt connect -> wait 6_000 milliseconds -> and so on
+        /// </summary>
+        public ReconnectionPolicy()
         {
-            _attemptsMade = 0;
-            _minReconnectInterval = _reconnectStepInterval;
-            _maxAttempts = _initMaxAttempts;
+            reconnectStepInterval = 3_000;
+            currentReconnectInterval = reconnectStepInterval;
+            maxReconnectInterval = 30_000;
+            maxAttempts = null;
+            initMaxAttempts = null;
+            attemptsMade = 0;
         }
 
-        public void SetAttemptsMade(int count) => _attemptsMade = count;
-
-        public ReconnectionPolicy(int minReconnectInterval, int maxReconnectInterval, int? maxAttempts)
+        /// <summary>
+        ///     the <see cref="TwitchLib.Communication.Clients.TcpClient"/> or <see cref="TwitchLib.Communication.Clients.WebSocketClient"/>
+        ///     attempts to reconnect for <paramref name="maxAttempts"/> times
+        ///     <br></br>
+        ///     <br></br>
+        ///     with each attempt, the reconnect interval increases by the amount of <paramref name="minReconnectInterval"/>
+        ///     until it reaches <paramref name="maxReconnectInterval"/>
+        ///     <br></br>
+        ///     <br></br>
+        ///     Example:
+        ///     <br></br>
+        ///     <paramref name="minReconnectInterval"/> = 3_000
+        ///     <br></br>
+        ///     <paramref name="maxReconnectInterval"/> = 30_000
+        ///     <br></br>
+        ///     try to connect -> couldnt connect -> wait 3_000 milliseconds -> try to connect -> couldnt connect -> wait 6_000 milliseconds -> and so on
+        /// </summary>
+        /// <param name="minReconnectInterval">
+        ///     minimum interval in milliseconds
+        /// </param>
+        /// <param name="maxReconnectInterval">
+        ///     maximum interval in milliseconds
+        /// </param>
+        /// <param name="maxAttempts">
+        ///     <see langword="null"/> means <b>infinite</b>; it never stops to try to reconnect
+        /// </param>
+        public ReconnectionPolicy(int minReconnectInterval,
+                                  int maxReconnectInterval,
+                                  int maxAttempts)
         {
-            _reconnectStepInterval = minReconnectInterval;
-            _minReconnectInterval = minReconnectInterval > maxReconnectInterval
+            reconnectStepInterval = minReconnectInterval;
```

Comment:
There was no reason to remove underscores (`_`) from those private fields. Having underscore in the private field is common practice in C#. This change forces you to using `this` keyword that wouldn't be otherwise necessary.

Reply:
yeah I am not that happy with some of the naming as well but that is sth I can fix as well if Christian wants me to do that after the PR is merged

Reply:
Agree, it can always be changed later. It just creates extra changes :) Nothing code-breaking though. 
And I hope you don't mind I joined the review.

Reply:
always happy for more pairs of eyes. especially on big PRs :)